rs/replicated_state/src/canister_state/queues/message_pool/tests.rs

```diff
@@ -132,36 +142,46 @@ fn test_take() {
             let request = request(deadline);
             let response = response(deadline);
 
-            // Insert the two messages.
-            let (request_id, response_id) = match context {
-                Context::Inbound => (
-                    pool.insert_inbound(request.clone().into()),
-                    pool.insert_inbound(response.clone().into()),
-                ),
-                Context::Outbound => (
-                    pool.insert_outbound_request(request.clone().into(), time(14).into()),
-                    pool.insert_outbound_response(response.clone().into()),
-                ),
-            };
-
-            let request: RequestOrResponse = request.into();
-            let response: RequestOrResponse = response.into();
-
-            // Ensure that the messages are now in the pool.
-            assert_eq!(Some(&request), pool.get(request_id));
-            assert_eq!(Some(&response), pool.get(response_id));
-
-            // Actually take the messages.
-            assert_eq!(Some(request), pool.take(request_id));
-            assert_eq!(Some(response), pool.take(response_id));
-
-            // Messages are gone.
-            assert_eq!(None, pool.get(request_id));
-            assert_eq!(None, pool.get(response_id));
-
-            // And cannot be taken out again.
-            assert_eq!(None, pool.take(request_id));
-            assert_eq!(None, pool.take(response_id));
+            match context {
+                Context::Inbound => {
+                    let request_id = pool.insert_inbound(request.clone().into());
+                    let response_id = pool.insert_inbound(response.clone().into());
+                    test_take_impl(request_id, response_id, request, response, &mut pool);
+                }
+                Context::Outbound => {
+                    let request_id =
+                        pool.insert_outbound_request(request.clone().into(), time(14).into());
+                    let response_id = pool.insert_outbound_response(response.clone().into());
+                    test_take_impl(request_id, response_id, request, response, &mut pool);
+                }
+            }
+
+            fn test_take_impl<T>(
```

Comment:
I guess it's a matter of taste but I usually define the function before I use it in cases like this.

Reply:
I did it this way to minimize the diff. But you're right, it is probably not worth the confusion.